wagtail/core/fields.py

```diff
@@ -90,13 +95,34 @@ def __init__(self, block_types, **kwargs):
 
         self.stream_block.set_meta_options(block_opts)
 
+    def _check_json_field(self):
+        if type(self.use_json_field) is not bool:
+            warnings.warn(
+                f"StreamField should explicitly set use_json_field argument to True/False instead of {self.use_json_field}.",
+                RemovedInWagtail219Warning,
+                stacklevel=3,
+            )
+
     def get_internal_type(self):
-        return "TextField"
+        return "JSONField" if self.use_json_field else "TextField"
+
+    def get_lookup(self, lookup_name):
+        if self.use_json_field:
+            return models.JSONField(encoder=DjangoJSONEncoder).get_lookup(lookup_name)
```

Comment:
Nitpick: I wonder whether it would be worth breaking out the idea of "getting the JSONField field object" into another method/property returning `models.JSONField(encoder=DjangoJSONEncoder)`, just so we make sure always to return a field with the same options set in the future

Reply:
Not nitpick at all! [I considered it before](https://github.com/wagtail/wagtail/pull/8039#discussion_r813921187), but I forgot to add it. Thanks for reminding 👍